concept/type/impl/TypeImpl.java

```diff
@@ -99,6 +100,11 @@ public long getInstancesCount() {
 
     @Override
     public void setLabel(String label) {
+        try {
+            TypeQL.parseLabel(label);
+        } catch (TypeQLException e) {
+            throw TypeDBException.of(e);
+        }
```

Comment:
This TypeQL parse method throws an exception when the label does not conform to the syntax definition.